tests/TestData/Identity.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MatchBot\Tests\TestData;
+
+/**
+ * Static Identity helpers for both unit & integration tests.
+ */
+class Identity
+{
+    /**
+     * @see self::getTestIdentityTokenIncomplete()
+     */
+    public static function getTestIdentityTokenComplete(): string
+    {
+        // As below but `"complete": true`.
+        return 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.' .
+            'eyJpc3MiOiJodHRwczovL3VuaXQtdGVzdC1mYWtlLWlkLXN1Yi50aGViaWdnaXZldGVzdC5vcmcudWsiLCJpYXQiOjE2NjM5NDQ4ODks' .
+            'ImV4cCI6MjUyNDYwODAwMCwic3ViIjp7InBlcnNvbl9pZCI6IjEyMzQ1Njc4LTEyMzQtMTIzNC0xMjM0LTEyMzQ1Njc4OTBhYiIsImNv' .
+            'bXBsZXRlIjp0cnVlLCJwc3BfaWQiOiJjdXNfYWFhYWFhYWFhYWFhMTEifX0.9zk7DUdvC9BWuRhXo2p7r12ZiTuREV7v9zsY97p_fyA';
+    }
+
+    public static function getTestIdentityTokenIncomplete(): string
+    {
+        // One-time, artifically long token generated and hard-coded here so that we don't
+        // need live code just for MatchBot to issue ID tokens only for unit tests.
+        // Token is for Stripe Customer cus_aaaaaaaaaaaa11.
+        //
+        // Base 64 decoded body part:
+        // {
+        //  "iss":"https://unit-test-fake-id-sub.thebiggivetest.org.uk",
+        //  "iat":1663436154,
+        //  "exp":2524608000,
+        //  "sub": {
+        //      "person_id":"12345678-1234-1234-1234-1234567890ab",
+        //      "complete":false,
+        //      "psp_id":"cus_aaaaaaaaaaaa11"
+        //  }
+        // }
+        $dummyPersonAuthTokenValidUntil2050 = 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwczovL3VuaXQtdGVz' .
+            'dC1mYWtlLWlkLXN1Yi50aGViaWdnaXZldGVzdC5vcmcudWsiLCJpYXQiOjE2NjM0MzYxNTQsImV4cCI6MjUyNDYwODAwMCwic3ViIjp7' .
+            'InBlcnNvbl9pZCI6IjEyMzQ1Njc4LTEyMzQtMTIzNC0xMjM0LTEyMzQ1Njc4OTBhYiIsImNvbXBsZXRlIjpmYWxzZSwicHNwX2lkIjoi' .
+            'Y3VzX2FhYWFhYWFhYWFhYTExIn19.KdeGTDkkWCjI4-Kayay0LKn9TXziPXCUxxTPIZgGxxE';
+
+        return $dummyPersonAuthTokenValidUntil2050;
```

Comment:
could we do the encoding at the time we run the tests, so that it will be possible to see what's going on here and edit without having to read the comments? Or at least add a comment about how to generate the encoded data.

I tried decoding it but it didn't work, not sure why.  There was a chunk of unreadable stuff at the end of the decoded string, and this makes test error with `Slim\Exception\HttpUnauthorizedException: Unauthorised`

```
        return base64_encode('{"typ":"JWT","alg":"HS256"}{"iss":"https://unit-test-fake-id-sub.thebiggivetest.org.uk","iat":1663436154,"exp":2524608000,"sub":{"person_id":"12345678-1234-1234-1234-1234567890ab","complete":false,"psp_id":"cus_aaaaaaaaaaaa11"}})׆L9$X(�লk-
```

Reply:
And if we can't make this do the encoding at runtime then for clarity I'd suggest having two `public const string`s instead of public static functions.

Reply:
@bdsl If you're trying to replace a whole token then you need a header, body and signature. https://developer.atlassian.com/cloud/bitbucket/understanding-jwt-for-apps/#structure-of-a-jwt-token

I don't think there is a particularly easy way to set up tests to make Identity tokens without giving this app more 'real' runtime secrets than it needs, which is why I went for a long lived one (some time ago, so I also don't really remember)

Reply:
I don't follow - base64 is a non-cryptographic and reversible encoding that doesn't require any secret to run. So if we need a base64 encoded value why wouldn't we be able to generate it at runtime. Can chat tomorrow if you have time.

Reply:
@bdsl Per the link above, a JWT has 3 separately base64'd parts, the last of which is a signature which requires knowledge of a secret.

Reply:
I still don't get it I'm afraid. Surely if we can base64 encode something then anyone who can read it can decode it, since there's no secret involved in base64. And if they can decode it why not keep it decoded in the first place? A secret is needed to compute the HMAC but that's a separate step to the base64 encoding - the key needs to be secret, the HMAC can't be secret.

There's also a separate question about why we would need to use a real secret at all in tests - we should be able to use an alternate fake secret that is only for test purposes and doesn't actually need to be kept secret.

But I don't think the question about DX of these tests files is important enough to hold up merging this further now.

Reply:
ah I'm a bit less confused now - I was trying to decode it as a whole which is why it doesn't work, it has to be decoded one part at a time in base64, or as a whole by pasting into a tool like https://jwt.io/ 

But it looks like the 'secret' we're using here is not a real secret at all and there's no reason for it to be- it's already just "unitTestJWTSecret" and it's already in the repo at https://github.com/thebiggive/matchbot/blob/dea3e98984e3d12a43560525e2a13b08655e431e/phpunit.xml#L41

Reply:
Ah, fair enough! Yep, agree it would be better DX if this was done in code but will go ahead and merge what we have so far for today.